Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/SharedLogic/DriverPool/Config.hs

```diff
@@ -34,13 +36,37 @@ data CancellationScoreRelatedConfig = CancellationScoreRelatedConfig
 getDriverPoolConfig ::
   (CacheFlow m r, EsqDBFlow m r) =>
   Id Merchant ->
+  Variant.Variant ->
   Meters ->
   m DriverPoolConfig
-getDriverPoolConfig merchantId dist = do
+getDriverPoolConfig merchantId vehicleVariant dist = do
   configs <- CDP.findAllByMerchantId merchantId
-  let applicableConfig = find filterByDist configs
+  let applicableConfig = find filterByDistAndDveh configs
   case configs of
     [] -> throwError $ InvalidRequest "DriverPoolConfig not found"
     (config : _) -> pure $ maybe config identity applicableConfig
+  where
+    filterByDistAndDveh cfg = dist >= cfg.tripDistance && cfg.vehicleVariant == vehicleVariant
```

Comment:
Make `vehicleVariant` in config as Maybe field for backward compatibility. If vehicleVariant entry is not there in config, just return the values with null vehicle variant.